cpp_linter/cli.py

```diff
@@ -304,6 +305,31 @@
 )
 
 
+def _parse_jobs(val: str) -> Optional[int]:
+    try:
+        jobs = int(val)
+    except ValueError:  # pragma: no cover
+        raise argparse.ArgumentTypeError(f"invalid value: {val} (must be an integer)")
```

Comment:
We can actually cover this in a unit test using `pytest.mark.xfail` (see other review comment). TBH, I think this whole function could be reduced to a lambda:
```diff
-    type=_parse_jobs,
+    type=lambda x: None if int(x) <= 0 else int(x),
```
But maybe its not good to rely on `int()` raising a generic looking `ValueError`.

Instead, we can raise `from` the caught `ValueError` to avoid a lengthy traceback output:
```suggestion
    except ValueError as exc:
        raise argparse.ArgumentTypeError(
            f"Invalid -j (--jobs) value: {val} (must be an integer)"
        ) from exc
```